sdk/servicebus/Azure.Messaging.ServiceBus/samples/DeadLetterQueue/Program.cs

```diff
@@ -0,0 +1,301 @@
+            // run a receive loop that explicitly pushes messages into the DLQ when 
+            // they don't satisfy a processing condition. The fix-up receive loop inspects 
+            // the DLQ, fixes the "faulty" messages, and resubmits them into processing. 
+            var sendTask = SendMessages(sender, int.MaxValue);
+            var receiveTask = ReceiveMessages(queueName, cts.Token);
+            var fixupTask = PickUpAndFixDeadletters(queueName, sender, cts.Token);
+
+            // wait for a key press or 10 seconds
+            await Task.WhenAny(
+                Task.Run(() => Console.ReadKey()),
+                Task.Delay(TimeSpan.FromSeconds(10))
+            );
+
+            // end the processing 
+            cts.Cancel();
+            // await shutdown and exit
+            await Task.WhenAll(sendTask, receiveTask, fixupTask);
+        }
+
+        private static Task SendMessages(ServiceBusSender sender, int maxMessages)
+        {
+            dynamic data = new[]
+            {
+                new {name = "Einstein", firstName = "Albert"}, new {name = "Heisenberg", firstName = "Werner"},
+                new {name = "Curie", firstName = "Marie"}, new {name = "Hawking", firstName = "Steven"},
+                new {name = "Newton", firstName = "Isaac"}, new {name = "Bohr", firstName = "Niels"},
+                new {name = "Faraday", firstName = "Michael"}, new {name = "Galilei", firstName = "Galileo"},
+                new {name = "Kepler", firstName = "Johannes"}, new {name = "Kopernikus", firstName = "Nikolaus"}
+            };
+
+            // send a message for each data entry, but at most maxMessages
+            // we're sending in a loop, but don't block on each send, but 
+            // rather collect all sends in a list and then wait for all of
+            // them to complete asynchronously, which is much faster  
+            var tasks = new List<Task>();
+            for (int i = 0; i < Math.Min(data.Length, maxMessages); i++)
+            {
+                // each message has a JSON body with one of the data rows
+                var message = new ServiceBusMessage(new BinaryData(jsonSerializable: data[i]))
+                {
+                    ContentType = "application/json", // JSON data
+                    Subject = i % 2 == 0 ? "Scientist" : "Physicist", // random picked header
+                    MessageId = i.ToString(), // message-id 
+                    TimeToLive = TimeSpan.FromMinutes(2) // message expires in 2 minutes 
+                };
+
+                // start sending this message
+                lock (Console.Out)
+                {
+                    Console.ForegroundColor = ConsoleColor.DarkYellow;
+                    Console.WriteLine($"Message sent: Id = {message.MessageId}");
+                    Console.ResetColor();
+                }
+
+                ;
+                tasks.Add(sender.SendMessageAsync(message).ContinueWith((t) =>
```

Comment:
We should consider commenting on the `ContinueWith` use;  I don't think that's a widely understood pattern.